packages/application-shell/src/components/locale-switcher/locale-switcher.tsx

```diff
@@ -79,22 +105,45 @@ const LocaleSwitcher = (props: Props) => {
         name="locale-switcher"
         aria-labelledby={LOCALE_SWITCHER_LABEL_ID}
         onChange={handleSelection}
-        options={props.availableLocales.map((locale) => ({
-          label: locale,
-          value: locale,
-        }))}
+        options={localeOptions}
         components={{
           SingleValue,
           ValueContainer: PatchedValueContainer,
           MenuList: CustomMenuList,
+          GroupHeading: (groupProps) => (
+            <CustomGroupHeading
+              {...groupProps}
+              setIsOpen={setIsDescriptionDialogOpened}
+            />
+          ),
         }}
         isClearable={false}
         backspaceRemovesValue={false}
         isSearchable={false}
         horizontalConstraint={'auto'}
         appearance="quiet"
         maxMenuHeight={360}
+        minMenuWidth={3}
       />
+      {/* Dialog that explains the locales */}
+      <InfoDialog
+        isOpen={isDescriptionDialogOpened}
+        title="Selecting a data locale"
```

Comment:
This also needs to be moved to a message

Reply:
Done here - https://github.com/commercetools/merchant-center-application-kit/pull/3403/commits/03b02da2ba93be1689de7f2f6e29901902cffa22